common/Linux/ScreenSaverDBus.cpp

```diff
@@ -0,0 +1,109 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2010  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <dbus/dbus.h>
+#define BUS_NAME "org.freedesktop.ScreenSaver"
+#define BUS_PATH "/org/freedesktop/ScreenSaver"
+#define BUS_INTERFACE "org.freedesktop.ScreenSaver"
+static DBusConnection* connection = nullptr;
+static dbus_uint32_t cookie;
+bool ChangeScreenSaverStateDBus(bool inhibit_requested = true, char* program_name = (char*)"Unknown", char* reason = (char*)"Unknown")
+{
+	//Initialize a DBusError var.
+	DBusError error_dbus;
+	dbus_error_init(&error_dbus);
+	//Initialize connection.
+	//With dbus_bus_get() If a connection to the bus already exists, then that connection is returned.
+	connection = dbus_bus_get(DBUS_BUS_SESSION, &error_dbus);
+	if (!connection || dbus_error_is_set(&error_dbus))
+	{
+		dbus_error_free(&error_dbus);
+		dbus_connection_unref(connection);
+		return false;
+	}
+	char* bus_method;
```

Comment:
wouldn't it be nicer to use a std::string and declare it like

std::string bus_method = (inhibit_requested) ? "Inhibit" : "UnInhibit";

then on the call you can do

DBusMessage* message = dbus_message_new_method_call(BUS_NAME, BUS_PATH, BUS_INTERFACE, bus_method.c_str());